File: src/de/blackpinguin/android/sindwirschonda/si/SITime.java
package de.blackpinguin.android.sindwirschonda.si;

import java.util.HashMap;

public class SITime extends SIUnit {
	
	private static HashMap<String, SIUnit> cache = new HashMap<String, SIUnit>();
	
	public static SIUnit find(String abbreviation){
		SIUnit r = cache.get(abbreviation);
		
		if(r == null)
			return NaU;
		else return r;
	}
	
	private SITime(String name, String abbreviation, double baseUnitMultiplier){
		super(name, abbreviation, baseUnitMultiplier);
		cache.put(abbreviation, this);
	}
	
	public static final SITime s = new SITime("second", "s", 1.0);
	public static final SITime m = new SITime("minute", "m", 60.0);
	public static final SITime h = new SITime("hour", "h", m.getBaseUnitMultiplier() * 60.0);
	public static final SITime d = new SITime("day", "d", h.getBaseUnitMultiplier() * 24.0);
	public static final SITime a = new SITime("annus", "a", d.getBaseUnitMultiplier() * 365.25);
	
	protected SIUnit getBaseUnit(){
		return s;
	}
	
	public SIUnit multiply(SIUnit other){
		if(other instanceof SISpeed)
			return SIDistance.m;
		return NaU;
	}
	
	public SIUnit divide(SIUnit other){
		return NaU;
	}
}

File: src/de/blackpinguin/android/sindwirschonda/si/SITime.scala
package de.blackpinguin.android.sindwirschonda.si

import de.blackpinguin.android.sindwirschonda.R

//Begleitobjekt
object SITime extends SIUnitType[SITime] {

  def nameID = R.string.Time
  def arrayID = R.array.time

  val s = SITime(0, 1.0)
  val m = SITime(1, 60.0)
  val h = SITime(2, m.baseUnitMultiplier * 60.0)
  val d = SITime(3, h.baseUnitMultiplier * 24.0)
  val a = SITime(4, d.baseUnitMultiplier * 365.25)

}

//Klasse
case class SITime private (_abbr: String, _bum: Double) extends SIUnit(_abbr, _bum) {

  def *(other: SIUnit) = other match {
    case _: SISpeed => SIDistance.m
    case _ => SIUnit.NaU
  }

  def /(other: SIUnit) = SIUnit.NaU

}